packages/asset-service/src/service/AssetService.ts

```diff
@@ -1,9 +1,64 @@
+import { caip19 } from '@shapeshiftoss/caip'
 import { Asset, AssetDataSource, BaseAsset, ChainTypes, NetworkTypes } from '@shapeshiftoss/types'
 import axios from 'axios'
 
+/* eslint-disable */
+const Identicon = require('identicon.js')
+const { fonts, renderPixels } = require('js-pixel-fonts')
+/* eslint-enable */
+
 import assetsDescriptions from './descriptions.json'
 import localAssetData from './generatedAssetData.json'
 
+const getRenderedIdenticonB64 = (symbol: string, tokenId: string): string => {
+  // `imageSize` defines the size (px) of generated identicon. It can be modified according to needs.
+  const imageSize = 128
+  // `symbolScale` controls the scale of the pixelSymbol as its original size (5 * 7) is too small.
+  // e.g. applying `symbolScale` = 12 makes the size of pixelSymbol 60 * 84
+  const symbolScale = 12
+  // options. checks https://github.com/stewartlord/identicon.js
+  const options = {
+    background: [45, 55, 72, 255],
+    size: imageSize
+  }
+  // In order to additionally render text, we acquire PNGLib object here which has some convenient png operator.
+  // https://github.com/stewartlord/identicon.js/blob/master/pnglib.js
+  const identiconImage = new Identicon(tokenId?.substring(2), options).render()
+
+  // we just return if no symbol
+  if (symbol.length === 0) return `data:image/png;base64,${identiconImage.getBase64()}`
+
+  // here we start to render the text
+  // we only take the first letter from the symbol to render
+  const symbolPixels = renderPixels(symbol[0], fonts.sevenPlus)
```

Comment:
In the issue, theobold says that `The display of the icon should include up to 5 characters of the asset's symbol.`

Did I missed something then?

Reply:
Yes, that is indeed a requirement from the issue. However, during my testing I find the text might be too small if we need to render 5 characters. Here is the example:

Render 5 characters, scale = 4:
![Render 5 characters:](https://user-images.githubusercontent.com/32592090/154828160-a37c1a11-c6b0-42bf-810d-89fce17a3f37.png)

Render 3 characters, scale = 7:
![Render 3 characters:](https://user-images.githubusercontent.com/32592090/154828161-ae8d9cb4-9004-49cc-8e90-97dd52177e38.png)

Render 1 characters, scale = 12:
![Render 1 characters:](https://user-images.githubusercontent.com/32592090/154828168-eed12a32-abb0-48a1-8765-84cbb3c16d9f.png)

So I concluded that rendering 1-3 characters might be most suitable for better UX. Anyway, in the latest commit we will render 3 characters instead of 1.

Reply:
And the text now is also configurable so that whether to render 1 character or 5 characters can be decided by the downstream applications.